packages/flutter/lib/src/material/colors.dart

```diff
@@ -2,7 +2,76 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:ui' show Color;
+import 'dart:ui' show Color, hashValues;
+
+class _ColorSwatch extends Color {
+  const _ColorSwatch(int primary, this.swatch) : super(primary);
+
+  /// The shades of this color defined by material design.
+  final Map<int, Color> swatch;
+
+  /// Returns an element of the [swatch] table.
+  Color operator [](int index) => swatch[index];
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! _ColorSwatch)
+      return false;
+    final _ColorSwatch typedOther = other;
+    return super==(other) && swatch == typedOther.swatch;
+  }
+
+  @override
+  int get hashCode => hashValues(runtimeType, value, swatch);
+
+  @override
+  String toString() => '$runtimeType(primary value: ${super.toString()})';
+
+}
+
+/// Defines a single color as well a color swatch with ten shades of the color.
+///
+/// The color's shades are referred to by index. The greater the index, the
+/// darker the color. There are 10 valid indices: 50, 100, 200, ..., 900.
+/// The value of this color is the same the value of index 400, or [shade400].
```

Comment:
looks like you didn't update the dartdoc here.

Also, I think we should rephrase this to say that the value of the color _should_ be the same as the value of index 500, because nothing actually ensures it, so someone could easily create a version that wasn't like that.